// @config/config.ts

import { models } from "@db/models/models"

const bunyan = require('bunyan');

const loggerOptions = {
  name: "ndtech-sapper",
  src: true,
  streams: [
      {
          stream: process.stdout            // log INFO and above to stdout
      },
      {
          path: './dev-prod-db-error.log'  // log ERROR and above to a file
      }
  ],
  serializers: {
      app: function (app) {
          if (app) {
              return {
                  appName: app.name,
                  appUrl: app.url
              };
          }
      },
      req: bunyan.stdSerializers.req,
      res: bunyan.stdSerializers.res
  }
};

const initGlobalConfig = function () {
  var config = {
      dbName: "ndtech-sapper",
      models: models,
      loggerOptions: loggerOptions,
      logger: bunyan.createLogger(loggerOptions),
      dbPath: process.env.APP_DB_PATH ? process.env.APP_DB_PATH : "mongodb://127.0.0.1/ndtech-sapper",
      dbOptions: {
        // useUnifiedTopology: true,
        useNewUrlParser: true,
        authSource: "ndtech-sapper",
        user: "ndtech-sapper",
        pass: process.env.MONGO_PASSWORD
      },
      db: null
  };
  // const models = require(path.join(__dirname, 'node_modules/@db/models'));

  // initServerSettings(config);
  // initDbSettings(config);
  // config.logger = setupLogger(config);
  // initGlobalConfigFiles(config, assets);
  // initAwsSettings(config);

  return config;
};


export let config = initGlobalConfig();
